src/internal/operators/switchMap.ts

```diff
@@ -133,10 +133,13 @@ class SwitchMapSubscriber<T, R> extends OuterSubscriber<T, R> {
     if (innerSubscription) {
       innerSubscription.unsubscribe();
     }
-    const innerSubscriber = new InnerSubscriber(this, undefined, undefined);
+    const innerSubscriber = new InnerSubscriber(this, value, index);
     const destination = this.destination as Subscription;
     destination.add(innerSubscriber);
-    this.innerSubscription = subscribeToResult(this, result, value, index, innerSubscriber);
+    this.innerSubscription = subscribeToResult(this, result, undefined, undefined, innerSubscriber);
+    if (this.innerSubscription !== innerSubscriber) {
```

Comment:
Can we add comments in this section explaining why we're doing what we're doing? Otherwise, there's a lot of mental math for the next developer.